src/wasm/encoder.cc

```diff
@@ -58,13 +58,19 @@ struct WasmFunctionBuilder::Type {
   LocalType type_;
 };
 
-WasmFunctionBuilder::WasmFunctionBuilder(Zone* zone)
+WasmFunctionBuilder::WasmFunctionBuilder(Zone* zone, const std::string& name)
```

Comment:
Not gaining anything but it keeps the API little clean. Anyways, replaced it by directly passing const unsigned char* and length from AstRawString. Please see if this looks better. This will not let add function be called with static strings